sales_embargo/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSaleEXT(WebsiteSale):
+    @http.route(
+        ["/shop/checkout"], type="http", auth="public", website=True, sitemap=False
+    )
+    def checkout(self, **post):
+        order = request.website.sale_get_order()
+        # IF PUBLIC ORDER
+        if order.partner_id.id == request.website.user_id.sudo().partner_id.id:
+            country_code = request.session["geoip"].get("country_code")
```

Comment:
Yes. Maybe it would be better to have the check at this stage also for the logged-in users because they may enter a different shipping address and should be able to reach that stage.